topics/statistics/tutorials/fruit_360/tutorial.md

```diff
@@ -0,0 +1,491 @@
+>    - *"Select input type:"*: `tabular data`
+>        - *"Training samples dataset"*: Select `train_X_10` dataset
+>        - *"Choose how to select data by column:"*: `All columns`
+>        - *"Dataset containing class labels or target values"*: Select the OHE representation of `train_y_10` dataset
+>        - *"Choose how to select data by column:"*: `All columns`
+>    - Click *"Execute"*
+>
+>
+{: .hands_on}
+
+The training step generates 3 datasets. 1) accuracy of the trained model, 2) the trained model, downloadable as a zip file, and 
+3) the trained model weights, downloadable as an hdf5 file. These files are needed for prediction in the next step.
+
+### **Model Prediction**
+
+> ### {% icon hands_on %} Hands-on: Testing the model
+>
+> - {% tool [Model Prediction](toolshed.g2.bx.psu.edu/repos/bgruening/model_prediction/model_prediction/1.0.8.2) %}
+>    - *"Choose the dataset containing pipeline/estimator object"* : Select the trained model from the previous step.
+>    - *"Choose the dataset containing weights for the estimator above"* : Select the trained model weights from the previous step.
+>    - *"Select invocation method"*: `predict`
+>    - *"Select input data type for prediction"*: `tabular data`
+>    - *"Training samples dataset"*: Select `test_X_10` dataset
+>    - *"Choose how to select data by column:"*: `All columns`
+>    - Click *"Execute"*
+>
+{: .hands_on}
+
+The prediction step generates 1 dataset. It's a file that has predictions (0 to 9 for the predicted fruit/vegetable) for every image 
+in the test dataset.
+
+### **Extract the Label column from test_y_10**
+
+> ### {% icon hands_on %} Hands-on: Advanced Cut
+>
+> - {% tool [Advanced Cut](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0) %}
+>    - *"File to cut"* : Select `test_y_10`
+>    - *"Operation"* : Select `Keep`
+>    - *"Delimited by"*: Select `Tab`
+>    - *"Cut by"*: Select `fields`
+>    - *"List of fields"*: Select `Column: 3`
+>    - Click *"Execute"*
+>
+{: .hands_on}
+
+### **Create One-Hot Encoding (OHE) representation of test labels**
+
+> ### {% icon hands_on %} Hands-on: One-Hot Encoding
+>
+> - {% tool [To categorical](https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_to_categorical/sklearn_to_categorical/1.0.8.3) %}
+>    - *"Input file"* : Select the output of the previous step.
+>    - *"Does the dataset contain header?"* : Select `Yes`
+>    - *"Total number of classes"*: Select `10`
+>    - Click *"Execute"*
+>
+{: .hands_on}
+
+
+
+### **Machine Learning Visualization Extension**
+
+> ### {% icon hands_on %} Hands-on: Creating the confusion matrix
+>
+> - {% tool [Machine Learning Visualization Extension](toolshed.g2.bx.psu.edu/repos/bgruening/ml_visualization_ex/ml_visualization_ex/1.0.8.2) %}
+>    - *"Select a plotting type"*: `Confusion matrix for classes`
+>    - *"Select dataset containing the true labels"*": `test_y_10`
```

Comment:
I think one step is missing here - `Does the dataset contain header: true` because `test_y_10` contains header as `Label`. @kxk302 

One minor suggestion: Two steps - `Extract the Label column from test_y_10` and `Machine Learning Visualization Extension` are repeated for two datasets `test_y_10` and `train_y_10`. Can you merge these two steps using the `multiple datasets` feature for the two tools (`Advancted cut` and `to categorical`)? Also, why do we have the test_y_10 dataset represented as OHE, it seems we don't use it? Thanks!

Apart from these comments, I am good :)

Reply:
@anuprulez I added 'Does the dataset contain header: true' for test_y_10 in ML Visualization Extension. Thanks for catching it!

The Extract Label and ML Visualization Extension only appear once. Extract Label for train_y_10 and Visualization Extension for y_test_10. Just so I'm clear where are you seeing them being repeated? Thx

Reply:
Never mind @anuprulez, I found the duplicate Extract Label and OHE of test_y_10 and removed it.

Reply:
@kxk302 sorry, not for `Machine Learning Visualization Extension` section but only for `Extract the Label column` and `Create One-Hot Encoding` sections - we can use them, once each, by activating `multiple datasets` options in the respective tools. So, we can use `train_y_10` and `test_y_10` datasets as multiple datasets input to `Advancted cut` tool and similarly for the `to categorical` tool. Thanks!